### PR TITLE
BUGFIX SIO-2319 add applications api

### PR DESCRIPTION
**Pull request by acidmk**

### CODE REVIEW

File: applications.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+type ApplicationsService service
+
+type LoginLink struct {
+	Type    string `json:"type"`
```
Comment (dshemin) [inline applications.go:12]: Please add an enum for the type.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: applications.go (hunk 1 above)
Comment (acidmk) [reply applications.go]: done

File: applications.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+type ApplicationsService service
+
+type LinkType string
```
Comment (dshemin) [inline applications.go:11]: Please rename to `LoginLinkType`

File: applications.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+const (
+	None    LinkType = "none"
+	URL     LinkType = "url"
+	JS_CODE LinkType = "js_code"
```
Comment (dshemin) [inline applications.go:16]: Should be `LoginLinkTypeJSCode`. Use this as an example https://github.com/solusio/solus-go-sdk/blob/master/icon.go#L7-L11

**Review by dshemin — CHANGES_REQUESTED** [review]

File: applications.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+type ApplicationsService service
+
+type LoginLinkType string
+
+const (
+	LoginLinkNone   LoginLinkType = "none"
+	LoginLinkURL    LoginLinkType = "url"
+	LoginLinkJSCode LoginLinkType = "js_code"
```
Comment (dshemin) [inline applications.go:16]: Sorry `LoginLinkTypeJSCode`. Main thing is to repeat a type name in the constant 'cause Golang doesn't have enum it's a common way to name like that.

**Review by dshemin — CHANGES_REQUESTED** [review]

**Review by dshemin — APPROVED** [review]
LGTM!

**Pull request merged**